### ISSUE TITLE
export 'onServerPrefetch' was not found in 'vue-demi'

### ISSUE THREAD

**Issue by micbra** [vuejs/vue-apollo#1102]
**Describe the bug**
`@vue/apollo-composable` doesn't work with vue 3 since `vue-demi` cant resolve `onServerPrefetch`. Seems it's only provided by `@vue/composition-api`

**To Reproduce**
1. Create a new vue 3 project
2. Install `graphql graphql-tag @apollo/client` as described [here](https://v4.apollo.vuejs.org/guide/installation.html#manual-installation)
3. Install and configure `@vue/apollo-composable` as described [here](https://v4.apollo.vuejs.org/guide-composable/setup.html#_1-install-vue-apollo-composable)
4. See warning in terminal:
```
 warning  in ./node_modules/@vue/apollo-composable/dist/useQuery.js

"export 'onServerPrefetch' was not found in 'vue-demi'
```

**Expected behavior**
One should be able to import `onServerPrefetch` from `@vue/apollo-composable` (or directly from vue?)

**Versions**
vue: 3.0.2
@vue/apollo-composable: 4.0.0-apha.12
@apollo/client: 3.2.7

**Additional context**
Trying to import `onServerPrefetch` from `@vue/composition-api` leads to another warning on the terminal:
```
 warning  in ./node_modules/@vue/composition-api/dist/vue-composition-api.esm.js

"export 'default' (imported as 'Vue') was not found in 'vue'
```

**Comment by Skulsta** [comment]
Any update on this?

**Comment by Gyurmatag** [comment]
Can you please provide an update on this issue?

**Comment by vjself** [comment]
Any updates anywhere on this issue yet?

**Comment by camwhite** [comment]
Anything on this yet?

**Comment by tbarkley** [comment]
Updates please?

**Comment by victororlyk** [comment]
have the same issue

**Comment by killix** [comment]
I am working with this hack for now https://github.com/killix/vue-apollo/commit/1dbb4bf53a7ac68d72e67a71c14bd9beacd67839

**Comment by camwhite** [comment]
> I am working with this hack for now [killix@1dbb4bf](https://github.com/killix/vue-apollo/commit/1dbb4bf53a7ac68d72e67a71c14bd9beacd67839)

@killix does this break prefetching?

**Comment by killix** [comment]
Should not as vue-demi doesn't export the function.

**Comment by leonardoks16** [comment]
Same problem here

**Comment by tbarkley** [comment]
Very nice work btw but I'm having the same issue, any updates?

**Comment by leonardoks16** [comment]
Check this out, i alread tested with Vue, Vite.js and Node. [Astrid-Client](https://github.com/leonardoks16/astrid-clientv)

**Comment by abdullahseba** [comment]
Same issue.

**Comment by ronny1020** [comment]
Same problem here.

**Comment by JoJk0** [comment]
same here :<

**Comment by simultsop** [comment]
looking forward

**Comment by Tangel** [comment]
You can try install the latest `@vue/composition-api`, It work for me.

**Comment by guillaumeduhan** [comment]
Same here ✨

**Comment by sephirothwzc** [comment]
Same here ✨

**Comment by georgep11** [comment]
I am having the same issue

**Comment by JasonAtallah** [comment]
Any updates on this?

**Comment by nshusr** [comment]
patch fix in postinstall  https://github.com/vuejs/vue-apollo/issues/1011#issuecomment-679298995

**Comment by iamgmd** [comment]
Same here

**Comment by napalmpapalam** [comment]
same

**Comment by alex-bluetrain** [comment]
same thing

**Comment by LeBenLeBen** [comment]
Please stop posting comments saying you have the same issue. You can use the 👍 reaction on the initial post for that. I’m also a user that want to see this fixed but this won’t change anything except pissing off maintainers. Many people are getting notified and that’s just useless noise. Thank you 🙇🏻‍♂️

**Comment by blay12cedric** [comment]
I have found something interesting that you can use to handle this issue :

- Use `import { provide } from "vue";` instead of `import { provide } from "@vue/composition-api"`
- This `provide` will be used to set your apollo clients: 
 `provide(ApolloClients, {
      default: apolloClient,
    });`

I hope this will resolve your issue.

**Comment by napalmpapalam** [comment]
> I have found something interesting that you can use to handle this issue :
> 
> * Use `import { provide } from "vue";` instead of `import { provide } from "@vue/composition-api"`
> * This `provide` will be used to set your apollo clients:
>   `provide(ApolloClients, { default: apolloClient, });`
> 
> I hope this will resolve your issue.

I'm using `import { provide } from "vue";` but it doesn't help :(

like this: 
`import { DefaultApolloClient } from '@vue/apollo-composable'`
`import { apolloClient } from '@/apollo-client'`
`import { createApp, provide } from 'vue'`

`const app = createApp({`
`  setup () {`
`    provide(DefaultApolloClient, apolloClient)`
`  },`
`  render: () => h(App),`
`})`

**Comment by ant1m4tt3r** [comment]
Same problem in here

**Comment by notngan** [comment]
this solved the problems for me:
https://github.com/quasarframework/app-extension-apollo/issues/51#issuecomment-791977057

**Comment by boboldehampsink** [comment]
Vue 3.1.0 beta is out now with support for onServerPrefetch, see https://github.com/vuejs/vue-next/blob/master/CHANGELOG.md. If you use 3.1 beta, the issue will dissappear.

**Comment by cblaettl** [comment]
> Vue 3.1.0 beta is out now with support for onServerPrefetch, see https://github.com/vuejs/vue-next/blob/master/CHANGELOG.md. If you use 3.1 beta, the issue will dissappear.

Thanks for the heads-up, but now I'm getting the following error:

```
ReferenceError: getCurrentInstance is not defined
    at useQueryImpl (useQuery.ts:140)
    at useQuery (useQuery.ts:127)
```

I'm currently using vite, but I might switch back to webpack if this solves the issue. Anyone else experiencing the same issue?
My dependencies:

```
@apollo/client: 3.3.16
@vue/apollo-composable: 4.0.0-alpha.12
vue: 3.1.0-beta.2
```

**Edit:** Seems to work fine with Webpack. So I guess I stick to Webpack until this issue in vite is resolved.

**Comment by saibbyweb** [comment]
> Vue 3.1.0 beta is out now with support for onServerPrefetch, see https://github.com/vuejs/vue-next/blob/master/CHANGELOG.md. If you use 3.1 beta, the issue will dissappear.

` { 
"vue": "^3.1.0-beta.6"
}`

solves the issue for me (using vite)

**Comment by rickyvadala** [comment]
Guys great news!
I got it running, I realized that the documentation on the repo is not complete
https://github.com/quasarframework/app-extension-apollo/tree/v2

**Is missing this part:**

Your src/App.vue should be like this

```
<template>
  <router-view />
</template>
<script lang="ts">
  import { defineComponent, provide } from 'vue'
  import { ApolloClients } from '@vue/apollo-composable'
  import { apolloClients } from 'src/extensions/apollo/boot'

  export default defineComponent({
    name: 'App',
    setup() {
      provide(ApolloClients, apolloClients)
    },
  })
</script>
```

PD: Found this on the README of the library in your-app\node_modules\@quasar\quasar-app-extension-apollo\README.md

**Comment by jesusgabri3l** [comment]
> 
> 
> this solved the problems for me:
> [quasarframework/app-extension-apollo#51 (comment)](https://github.com/quasarframework/app-extension-apollo/issues/51#issuecomment-791977057)
Solved the issue!

**Comment by tronghieu60s** [comment]
> Guys great news!
> I got it running, I realized that the documentation on the repo is not complete
> https://github.com/quasarframework/app-extension-apollo/tree/v2
> 
> **Is missing this part:**
> 
> Your src/App.vue should be like this
> 
> ```
> <template>
>   <router-view />
> </template>
> <script lang="ts">
>   import { defineComponent, provide } from 'vue'
>   import { ApolloClients } from '@vue/apollo-composable'
>   import { apolloClients } from 'src/extensions/apollo/boot'
> 
>   export default defineComponent({
>     name: 'App',
>     setup() {
>       provide(ApolloClients, apolloClients)
>     },
>   })
> </script>
> ```
> 
> PD: Found this on the README of the library in your-app\node_modules@quasar\quasar-app-extension-apollo\README.md

It doesn't work for me.

**Comment by LeBenLeBen** [comment]
All you should have to do to fix this issue is to use Vue 3.1+
